api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -0,0 +1,266 @@
+#include "opentelemetry/trace/key_value_iterable.h"
+#include "opentelemetry/trace/propagation/http_text_format.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/span_context.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace trace
+{
+namespace propagation
+{
+static const nostd::string_view kTraceParent = "traceparent";
+static const nostd::string_view kTraceState  = "tracestate";
+static const int kVersionBytes               = 2;
+static const int kTraceIdBytes               = 32;
+static const int kSpanIdBytes                = 16;
+static const int kTraceFlagBytes             = 2;
+static const int kTraceDelimiterBytes        = 3;
+static const int kHeaderSize =
+    kVersionBytes + kTraceIdBytes + kSpanIdBytes + kTraceFlagBytes + kTraceDelimiterBytes;
+static const int kTraceStateMaxMembers    = 32;
+static const int kHeaderElementLengths[4] = {2, 32, 16, 2};
+
+// The HttpTraceContext provides methods to extract and inject
+// context into headers of HTTP requests with traces.
+// Example:
+//    HttpTraceContext.inject(setter,&carrier,&context);
+//    HttpTraceContext.extract(getter,&carrier,&context);
+template <typename T>
+class HttpTraceContext : public HTTPTextFormat<T>
+{
+public:
+  // Rules that manages how context will be extracted from carrier.
+  using Getter = nostd::string_view (*)(const T &carrier, nostd::string_view trace_type);
+
+  // Rules that manages how context will be injected to carrier.
+  using Setter = void (*)(T &carrier,
+                          nostd::string_view trace_type,
+                          nostd::string_view trace_description);
+
+  void Inject(Setter setter, T &carrier, const context::Context &context) noexcept override
+  {
+    SpanContext span_context = SpanContext();
+    GetCurrentSpan(context, span_context);
+    if (!span_context.IsValid())
+    {
+      return;
+    }
+    InjectImpl(setter, carrier, span_context);
+  }
+
+  context::Context Extract(Getter getter,
+                           const T &carrier,
+                           context::Context &context) noexcept override
+  {
+    SpanContext span_context    = ExtractImpl(getter, carrier);
+    nostd::string_view span_key = "current-span";
+    nostd::shared_ptr<Span> sp{new DefaultSpan(span_context)};
+    return context.SetValue(span_key, sp);
+  }
+
+  static void GetCurrentSpan(const context::Context &context, SpanContext &span_context)
+  {
+    const nostd::string_view span_key = "current-span";
+    context::Context ctx(context);
+    context::ContextValue span = ctx.GetValue(span_key);
+    if (nostd::holds_alternative<nostd::shared_ptr<Span>>(span))
+    {
+      span_context = nostd::get<nostd::shared_ptr<Span>>(span).get()->GetContext();
+    }
+  }
+
+  static void GenerateBuffer(nostd::string_view string, int bytes, uint8_t *buf)
+  {
+    const char *str_id = string.begin();
+    for (int i = 0; i < bytes; i++)
+    {
+      int tmp = HexToInt(str_id[i]);
```

Comment:
What if the input is not a hex string? (e.g. contains invalid character)

Reply:
If it is a invalid character it will return -1, which will be handled by repective functions using it, e.g trace id generator would create an all 0 trace id representing the invalidness of it.